qfieldsync/gui/cloud_login_dialog.py

```diff
@@ -120,15 +144,80 @@ def __init__(
+
     def on_rejected(self) -> None:
         QApplication.restoreOverrideCursor()
         if self.parent():
             self.parent().setEnabled(True)
             self.setEnabled(True)
 
     def toggle_server_url_visibility(self) -> None:
-        self.serverUrlLabel.setVisible(not self.serverUrlLabel.isVisible())
-        self.serverUrlCmb.setVisible(not self.serverUrlCmb.isVisible())
+        self.loginFormGroup.setVisible(not self.loginFormGroup.isVisible())
+
+    def clear_login_widgets(self) -> None:
+        self.set_login_groupbox_visibility(self.signInUsernameGroupBox, False)
+
+        # clear sso login buttons layout
+        while self.oauth2_buttons_layout.count():
+            child = self.oauth2_buttons_layout.takeAt(0)
+            if child.widget():
+                child.widget().deleteLater()
+
+    def set_login_groupbox_visibility(self, group_box: QGroupBox, visible: bool):
+        group_box.setEnabled(visible)
+
+    def fetch_server_auth_capabilities(self) -> None:
+        """
+        Fetches the provided server authentication method capabilities.
+        """
+        self.clear_login_widgets()
+        self.network_manager.set_url(self.serverUrlCmb.currentText())
+        self.auth_cap_reply = self.network_manager.get_auth_capabilities()
+        self.auth_cap_reply.finished.connect(self.on_fetch_auth_request_finished)
+
+    def on_fetch_auth_request_finished(self) -> None:
+        try:
+            auth_methods = self.network_manager.json_array(self.auth_cap_reply)
+        except CloudException as e:
+            QgsMessageLog.logMessage(
+                f"Could not fetch auth methods from server '{self.serverUrlCmb.currentText()}'",
+                "QFieldSync",
+                Qgis.Warning,
+            )
+            QgsMessageLog.logMessage(str(e), "QFieldSync", Qgis.Critical)
+            return
+
+        self.clear_login_widgets()
+
+        for auth_method in auth_methods:
+            # credentials login: enabled static groupbox.
+            if auth_method["id"] == "credentials":
+                self._credentials_auth_method = auth_method
+                self.set_login_groupbox_visibility(self.signInUsernameGroupBox, True)
+                continue
+
+            # sso provider: dynamically generate button to login.
+            self._sso_auth_methods[auth_method["id"]] = auth_method
+            login_button = QPushButton(
+                self.tr("Sign In with {provider}").format(provider=auth_method["name"])
+            )
+            login_button.setCursor(QCursor(Qt.CursorShape.PointingHandCursor))
+
+            # if google auth available, we need to display the logo for legal reasons.
+            if auth_method["id"] == "google":
+                login_button.setIcon(
+                    QIcon(
+                        os.path.join(
+                            os.path.dirname(__file__), "../resources/google.svg"
+                        )
+                    )
+                )
```

Comment:
This will need adjusting to align with the QFC PR here (https://github.com/opengisch/QFieldCloud/pull/1192).

It also seems like we will have no choice but to style the button itself (styling colors provided by QFC). You can use the QWidget's setStylesheet on the QPushButton[s] for that.